### PR TITLE
Feature: Item Stack Size

### PR DESCRIPTION
**Pull request by RayDeeUx**
Added:
- Abbreviations of Skyblock player stats
- Skyblock level
- Skill average
- collection % unlocked display
- collection maxed out % display
- recipes unlocked %
- bank tier
- dojo grades
- highest crafted minion tier
- wardrobe slot #s
- progress to next minion slot
- SO MUCH MORE

### CODE REVIEW

**Comment by RayDeeUx** [comment]
oh yeah i also added a regularname util function for item names bc haha yes

the player stat abbrvtns could use a bit of work tho, up to yall

File: src/main/java/at/hannibal2/skyhanni/utils/ItemUtils.kt [hunk 1]
```diff
@@ -17,6 +17,7 @@ import java.util.*
 object ItemUtils {
 
     fun ItemStack.cleanName() = this.displayName.removeColor()
+    fun ItemStack.regularName() = this.displayName
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/ItemUtils.kt]: `itemStack.name` exists already

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/utils/ItemUtils.kt:20]: motherfuck—

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/utils/ItemUtils.kt]: fixed already

File: src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt [hunk 2]
```diff
@@ -60,7 +60,7 @@ class MenuItemDisplayOverlayAbiphone {
         if ((stackSizeConfig.contains(1)) && (chestName.contains("Abiphone"))) {
             if (item.name == null) return ""
             val lore = item.getLore()
-            if ((item.name?.startsWith("§f§")) && (lore.any { it.lowercase().contains("§cdo not disturb") }) && lore.any { it.lowercase().contains("enabled!") }) {
+            if ((item.name.startsWith("§f§")) && (lore.any { it.lowercase().contains("§cdo not disturb") }) && lore.any { it.lowercase().contains("enabled!") }) {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt]: Please don't use lowercase in that situation. Just change the text you compare to correct upper case letters.

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt]: ill do it to just this scenario; knowing hypixel admins theyre going to break the "SkyBlock Menu" capitalization for april fools day

**Comment by RayDeeUx** [comment]
i cant believe im saying this but i've run out of ideas so im un-drafting this

**Review by hannibal002 — DISMISSED** [review]
Disable fps heavy toggles by default (see discord post)

**Review by hannibal002 — CHANGES_REQUESTED** [review]
` if (lore.takeLast(2).first().removeColor().replace("Right click to ", "").contains("enable") || lore.last().removeColor().replace("Right click to ", "").contains("enable")) colorCode = "§c"`

Lines like those.. idk

File: src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+        PS: T'was all a joke. Just don't do stupid shit like
+        ` if (!(chestName == "Visitor's Logbook")) return "" `
+        and you *should* be fine for the most part.
+        ----------------------------------------------------------------------------
+        */
+
+        //NOTE: IT'S String.length, NOT String.length()!
+        
+        if ((stackSizeConfig.contains(0)) && (chestName.contains("Abiphone") || chestName.contains("AⒷiphone")) && (itemName == ("Contacts Directory"))) {
```
Comment (Helfull) [inline src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt]:
```
if ((stackSizeConfig.contains(0))
```

All those "magic numbers" make it hard to understand what is checked for

i suggest using an enum for those or constants, to understand what is checked better

Comment (Helfull) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt]:
```
(chestName.contains("Abiphone") || chestName.contains("AⒷiphone"))
```

I suggest refactoring this into a method, as there are many places this is used and checked for, and if in the future hypixel adds more variants, it is easier to adapt

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt]: > there are many places this is used

for now it's within the new files i created, but if it's any consolation i've condensed it into one boolean value within the offending file

as for adding it into lorenzutils, right now the one decent apiphone feature is in NEU and all i really added were stack sizes. we'll see though

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt]: > ```
> if ((stackSizeConfig.contains(0))
> ```
> 
> All those "magic numbers" make it hard to understand what is checked for
> 
> i suggest using an enum for those or constants, to understand what is checked better

on the one hand, enums for readability would be nice
on the other hand, that's using a thermonuclear bomb to skin a cat (and anyone who knows how to use enums—aka not me–would probably be willing to match the magic numbers to the corresponding option in the config)

**Comment by RayDeeUx** [comment]
moved to #694

**Pull request closed without merging**